Tests/Marketplace/marketplace_services.py

```diff
@@ -508,11 +522,34 @@ def add_pack_type_tags(self, yaml_content, yaml_type):
         Returns:
             Doesn't return
         """
+        logging.info("adding pack type tags")
         if yaml_type == 'Integration':
             if yaml_content.get('script', {}).get('feed', False) is True:
                 self._is_feed = True
             if yaml_content.get('script', {}).get('isfetchevents', False) is True:
                 self._is_siem = True
+            # this's the first integration in the pack, and the pack is in xsiem
+            if self._single_integration and 'marketplacev2' in self.marketplaces:
+
+                # the integration is not deprecated
+                if not yaml_content.get('deprecated', False):
+
+                    # the integration contains isfetch or isfetchevents
+                    if yaml_content.get('script', {}).get('isfetchevents', False) or \
+                            yaml_content.get('script', {}).get('isfetch', False) is True:
+                        logging.info(f"{yaml_content.get('name')} is a Data Source potential")
+                        self._is_data_source = True
+            # already has the pack as data source
+            elif not self._single_integration and self._is_data_source:
+
+                # got a second integration in the pack that's not deprecated
+                if not yaml_content.get('deprecated', False):
+                    logging.info(f"{yaml_content.get('name')} is no longer a Data Source potential")
+                    self._is_data_source = False
```

Comment:
This flag handling is a bit long, Please extract it to a new boolean function such as: `is_data_source_pack` and save the result back to `self._is_data_source`.